aqt/generate_combinations.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python3
+
+import json
+import logging
+import re
+from pathlib import Path
+from typing import (
+    Callable,
+    Dict,
+    Generator,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+from tqdm import tqdm
+
+from aqt.exceptions import ArchiveConnectionError, ArchiveDownloadError
+from aqt.helper import Settings
+from aqt.metadata import ArchiveId, MetadataFactory, Versions
+
+
+def is_blacklisted_tool(tool_name: str) -> bool:
+    for prefix in ("tools_qt3dstudio_",):
+        if tool_name.startswith(prefix):
+            return True
+    for suffix in ("_preview", "_early_access"):
+        if tool_name.endswith(suffix):
+            return True
+    return False
```

Comment:
This is a strange thing to do: it skips a large number of tools. The tools that this skips are not tracked by `aqt/combinations.json`, and they constitute the majority of the HTTP requests that this tool makes if they are not skipped. In the future, maybe this behavior should be turned off.